### PR TITLE
Investigate whether shipped Liberation Fonts are necessary

### PR DESCRIPTION
**Pull request by ntninja**
As part of #66, I proposed dropping `/share/calibre/fonts/liberation` from the Flatpak, since these files are already shipped (and apparently in a more recent version) in the FD runtime. In order to move forward with that request, it was decided to postpone the decision however.

Based on this I have some questions to @kovidgoyal:

  1. What are these files used for? Do they only act as a fallback to the system font resolution or are they referenced directly in some way that makes it essential to remain at the given path?
  2. If the former is true: Do you consider the presence of these files as essential for a complete Calibre distribution, even if it is guaranteed that a newer version of these fonts is always always available in the target environment?
  3. If the later is true: Would replacing them with a symlink (like Fedora and Debian apparently do) to their location in the platform image be considered an acceptable alternative to shipping them unmodified?

### CODE REVIEW

File: com.calibre_ebook.calibre.yaml [hunk 1]
```diff
@@ -30,13 +30,17 @@ modules:
...
+      - ln -s /usr/share/fonts/liberation-fonts/*.ttf "${FLATPAK_DEST}/lib/calibre/resources/fonts/liberation/"
+
       # Ensure that Calibre install script and the `xdg-utils` it uses will not
       # report errors because they are unable to find any suitable destination
       # directory to place their files into
       - mkdir -p "${FLATPAK_DEST}/bin" "${FLATPAK_DEST}"/share/{applications,desktop-directories,icons/hicolor,mime/packages}
 
       # Run Calibre system integration scripts
-      - /app/lib/calibre/calibre_postinstall --root="${FLATPAK_DEST}"
+      - '"${FLATPAK_DEST}/lib/calibre/calibre_postinstall" --root="${FLATPAK_DEST}"'
```
Comment (ghisvail) [inline com.calibre_ebook.calibre.yaml:43]: This change is unrelated to the PR.

**Review by ghisvail — CHANGES_REQUESTED** [review]

File: com.calibre_ebook.calibre.yaml (hunk 1 above)
Comment (Erick555) [reply com.calibre_ebook.calibre.yaml:43]: That's right although it's small cleanup change that makes manifest consistent and may not deserve separate PR which then will trigger another update for all users.

What concerns me a little is why it needs single quotation (`'`)while all other command in manifest don't?

Comment (ntninja) [reply com.calibre_ebook.calibre.yaml:43]: Because YAML otherwise gets confused about the double quote and throws a syntax error when it “unexpectedly” stops before the end of the line. (Did /me mention they are not a huge fan of YAML already? :wink:)

> That's right although it's small cleanup change that makes manifest consistent and may not deserve separate PR which then will trigger another update for all users.

Technically, if we merge 2 PRs in less than 3 hours (cool-down or whatever it was called), it will only be pushed as one update by Flathub. Practically speaking, I agree that this shouldn't need its own PR. Maybe if I split it into a separate commit instead?

Comment (Erick555) [reply com.calibre_ebook.calibre.yaml:43]: Does yaml needs double quotes and brackets at all (like shell script)? Currently manifest look quite chaotic when sometimes we have `"${FLATPAK_DEST}"` and sometimes `$FLATPAK_DEST`.

**Comment by ghisvail** [comment]
Le sam. 5 déc. 2020 à 18:20, Erick555 <notifications@github.com> a écrit :

> *@Erick555* commented on this pull request.
> ------------------------------
>
> In com.calibre_ebook.calibre.yaml
> <https://github.com/flathub/com.calibre_ebook.calibre/pull/70#discussion_r536826038>
> :
>
> >        # Ensure that Calibre install script and the `xdg-utils` it uses will not
>        # report errors because they are unable to find any suitable destination
>        # directory to place their files into
>        - mkdir -p "${FLATPAK_DEST}/bin" "${FLATPAK_DEST}"/share/{applications,desktop-directories,icons/hicolor,mime/packages}
>
>        # Run Calibre system integration scripts
> -      - /app/lib/calibre/calibre_postinstall --root="${FLATPAK_DEST}"
> +      - '"${FLATPAK_DEST}/lib/calibre/calibre_postinstall" --root="${FLATPAK_DEST}"'
>
> Does yaml needs double quotes and brackets at all (like shell script)?
>
Most of the time they are superfluous, apart from specific cases of string
interpolation.

I am happy with substituting /app with $FLATPAK_DEST in a separate commit,
however the extra quoting is likely unnecessary.

> Currently manifest look quite chaotic when sometimes we have
> "${FLATPAK_DEST}" and sometimes $FLATPAK_DEST.
>
The word you're looking for is inconsistent, not chaotic.

For the record, I am not against commits improving style, but it needs to
be separate from issues targeted by a PR to prevent unnecessary overhead
during review.

**Comment by ghisvail** [comment]
Thanks @ntninja
